config/policy-controller/deploy/local/kustomization.yaml

```diff
@@ -0,0 +1,10 @@
+# Local deployment overlay.
```

Comment:
This could be a nit: 
And not essential for this change. 

For a number of things I'd suggest some naming updates.
For example the 
- config/deploy is referring to gateway-controller
- config/policy-controller is referring to policy-controller
Also
- mgc-controller-manager is used for deployment/pod naming for gateway-controller
- mgc-policy-controller is used for deployment/pod naming for policy-controller

technically mgc-policy-controller could be expanded to multicluster-gateway-controller-policy-controller which could cause some confusion. maybe the prefix isn't appropriate anymore.

Reply:
> This could be a nit: And not essential for this change.
> 
> For a number of things I'd suggest some naming updates. For example the
> 
>     * config/deploy is referring to gateway-controller
> 
>     * config/policy-controller is referring to policy-controller

If it was intending to stay this way I'd agree, something like `config/policy-controller ` and `config/gateway-controller ` would make more sense, but moving the policy config was a first step and will hopefully soon be moved out of this repo entirely.

The policy-controller will eventually be it's own repo, and the current contents of `config/policy-controller` should end up being the contents of `config` in that repo and removed entirely from here. 

>       Also
> 
>     * mgc-controller-manager is used for deployment/pod naming for gateway-controller
> 
>     * mgc-policy-controller is used for deployment/pod naming for policy-controller
> 

I agree there were some naming inconsistencies brought in during the last couple of PRs (https://github.com/Kuadrant/multicluster-gateway-controller/pull/644, https://github.com/Kuadrant/multicluster-gateway-controller/pull/648, https://github.com/Kuadrant/multicluster-gateway-controller/pull/667) that need resolved. Not really my intention to try and fix all of that here though.

> 
> technically mgc-policy-controller could be expanded to multicluster-gateway-controller-policy-controller which could cause some confusion. maybe the prefix isn't appropriate anymore.

I'm not sure what this should end up being, but it should have a prefix i think that is added to all resources created as part of this deployment. Currently that is mgc.

Reply:
> > This could be a nit: And not essential for this change.
> > For a number of things I'd suggest some naming updates. For example the
> > ```
> > * config/deploy is referring to gateway-controller
> > 
> > * config/policy-controller is referring to policy-controller
> > ```
> 
> If it was intending to stay this way I'd agree, something like `config/policy-controller ` and `config/gateway-controller ` would make more sense, but moving the policy config was a first step and will hopefully soon be moved out of this repo entirely.
> 
> The policy-controller will eventually be it's own repo, and the current contents of `config/policy-controller` should end up being the contents of `config` in that repo and removed entirely from here.
> 

cool. no need for this change then. 

> > ```
> >   Also
> > 
> > * mgc-controller-manager is used for deployment/pod naming for gateway-controller
> > 
> > * mgc-policy-controller is used for deployment/pod naming for policy-controller
> > ```
> 
> I agree there were some naming inconsistencies brought in during the last couple of PRs (#644, #648, #667) that need resolved. Not really my intention to try and fix all of that here though.

yeah that's fair. I think it would be good to address them.  And it's not essential to fix them here. We could create a follow on issue about it? 

> 
> > technically mgc-policy-controller could be expanded to multicluster-gateway-controller-policy-controller which could cause some confusion. maybe the prefix isn't appropriate anymore.
> 
> I'm not sure what this should end up being, but it should have a prefix i think that is added to all resources created as part of this deployment. Currently that is mgc.